### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'path') 

### ISSUE THREAD

**Issue by AbdelghafourT** [HashLips/hashlips_art_engine#235]
Cannot read properties of undefined (reading 'path')
const image = await loadImage(`${_layer.selectedElement.path}`);

**Comment by Wilbur21** [comment]
Hi folks, it seems the v1.1.1 or earlier became touchy about characters in filenames. I see there is a comment "ensure there are no"-". Unfortunately replacing spaces with "-" is a standard feature of many softwares eg Photoshop and you need to be  a coder to change that. After I remove all the (hundreds) of "-", it still crashes because there is an "&" in the filenames placed there by the software my brother in Australia uses to create the images. It would be great if the software could run with legal filename characters. A run with a couple of files with the ampersand removed runs fine. See attached:
![Ampersand etc](https://user-images.githubusercontent.com/93607521/140838838-1ca5dd2e-15f6-41a0-8ff8-9b12e47b8396.png)
d.

**Comment by Wilbur21** [comment]
It seems the ampersand problem comes from the code in main.js line 246 which looks in strings for the "&".
Changing the "&" to ";" enabled the program to run with my filenames which contain "&"
Some other functionality is probably affected by this, possibly dnaBypass.
![Optional Query Parameters](https://user-images.githubusercontent.com/93607521/140907107-62f2ed0d-5852-4bd0-86d5-d2527e00ec1e.png)
.

**Comment by Wilbur21** [comment]
I found the "-" problem. It is being defined as the DNA_DELIMITER in main'js line 31.
I changed it to "^" and now the program runs with my filenames as I receive them from photoshop etc.
NB. I am not a skilled programmer (especially in these languages) and I have not tested ALL functionality of this great software - this is just a work-around to get my images composite
![DNA Delimiter](https://user-images.githubusercontent.com/93607521/140908886-56799125-7d2e-441e-a67c-a1364cc58b2f.png)
d.
